src/core/metaport.ts

```diff
@@ -0,0 +1,350 @@
+/**
+ * @license
+ * SKALE Metaport
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file metaport.ts
+ * @copyright SKALE Labs 2023-Present
+ */
+
+import { Provider, JsonRpcProvider, Contract } from 'ethers'
+
+import {
+  MetaportConfig,
+  TokenDataTypesMap,
+  Token,
+  TokenContractsMap,
+  TokenBalancesMap
+} from './interfaces'
+import { TokenType, TokenData, CustomAbiTokenType } from './dataclasses'
+
+import { getEmptyTokenDataMap } from './tokens/helper'
+import { getStepsMetadata } from '../core/transfer_steps'
+import { getChainEndpoint, initIMA, initMainnet, initSChain } from './network'
+import { MetaportState } from '../store/MetaportState'
+import { ERC_ABIS } from './contracts'
+
+import debug from 'debug'
+import { MainnetChain, SChain } from '@skalenetwork/ima-js'
+import { MAINNET_CHAIN_NAME } from './constants'
+
+const log = debug('ima:test:MainnetChain')
+
+export const createTokenData = (
+  tokenKeyname: string,
+  chainName: string,
+  tokenType: TokenType,
+  config: MetaportConfig
+): TokenData => {
+  const configToken: Token = config.connections[chainName][tokenType][tokenKeyname]
+  return new TokenData(
+    configToken.address,
+    tokenType,
+    tokenKeyname,
+    config.tokens[tokenKeyname],
+    configToken.chains,
+    chainName
+  )
+}
+
+export const addTokenData = (
+  tokenKeyname: string,
+  chainName: string,
+  tokenType: TokenType,
+  config: MetaportConfig,
+  tokens: TokenDataTypesMap
+) => {
+  tokens[tokenType][tokenKeyname] = createTokenData(tokenKeyname, chainName, tokenType, config)
+}
+
+export const createTokensMap = (
+  chainName1: string,
+  chainName2: string | null | undefined,
```

Comment:
Possible to reduce to this to null or undefined instead of both?

Reply:
upd: `null`